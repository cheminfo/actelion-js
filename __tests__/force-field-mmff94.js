'use strict';

const OCL = require('../core');

const { Molecule, ForceFieldMMFF94, ConformerGenerator } = OCL;

describe('ForceFieldMMFF94', () => {
  it('should generate force field', () => {
    const mol = Molecule.fromSmiles('COCCON');
    const gen = new ConformerGenerator(1);
    gen.getOneConformerAsMolecule(mol);
    const molfileBefore = mol.toMolfile();

    const ff = new ForceFieldMMFF94(mol, 'MMFF94');
    ff.minimise();
    const molfileAfter = mol.toMolfile();
    expect(molfileAfter).toMatchSnapshot();
    expect(molfileAfter).not.toBe(molfileBefore);
  });
});
