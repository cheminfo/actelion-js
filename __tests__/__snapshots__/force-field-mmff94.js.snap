// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForceFieldMMFF94 should generate force field 1`] = `
"
Actelion Java MolfileCreator 1.0

 15 14  0  0  0  0  0  0  0  0999 V2000
   -0.9921   -1.8622    5.7336 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.0258   -1.4185    4.7908 O   0  0  0  0  0  0  0  0  0  0  0  0
   -0.0734    0.0015    4.6137 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.0061    0.4348    3.6263 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.2873    0.3452    4.2659 O   0  0  0  0  0  0  0  0  0  0  0  0
    3.2975    0.8634    3.3480 N   0  0  0  0  0  0  0  0  0  0  0  0
   -2.0017   -1.6107    5.3948 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.9137   -2.9491    5.8217 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.8020   -1.4176    6.7152 H   0  0  0  0  0  0  0  0  0  0  0  0
   -1.0590    0.2723    4.2194 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.0820    0.4899    5.5825 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.0029   -0.2148    2.7436 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.8288    1.4740    3.3262 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.8044    1.5053    3.9592 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.8999    0.0444    3.2461 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  1  0  0  0  0
  3  4  1  0  0  0  0
  4  5  1  0  0  0  0
  5  6  1  0  0  0  0
  1  7  1  0  0  0  0
  1  8  1  0  0  0  0
  1  9  1  0  0  0  0
  3 10  1  0  0  0  0
  3 11  1  0  0  0  0
  4 12  1  0  0  0  0
  4 13  1  0  0  0  0
  6 14  1  0  0  0  0
  6 15  1  0  0  0  0
M  END
"
`;
